### PR TITLE
Add menu component syntactic sugar.

### PR DESCRIPTION
**Pull request by mateuszwyszynski**
This PR is response for #19 issue.

_uimenu_ creates a menu using a helper function _menuItem_. 

_menuItem_ creates one menu item/tab which is then added to the menu by _uimenu_

TODO: Check which features of a menu item are still missing implementation: dropdown, search, other (?)
TODO: Static item should be more flexible (not necessarily "header")
TODO: Submenus (e.g. in a vertical menu)

### CODE REVIEW

File: R/dsl.R [hunk 1]
```diff
@@ -103,3 +103,82 @@ generate_random_id <- function(prefix, id_length = 20) {
   random_id <- paste(sample(letters, id_length, replace = TRUE), collapse = "")
   paste0(prefix, "-", random_id)
 }
+
+create_hyperlink_item <- function(link = "", text = "", icon = "", style = "", item_feature = "") {
+  a(class = paste("item", item_feature),
+    href = link,
+    if (icon != "") {
```
Comment (filipstachura) [inline R/dsl.R]: can we extract:

```
icon_if_present <- function(icon) {
  if (icon != "") ...
}
```

File: R/dsl.R [hunk 2]
```diff
@@ -103,3 +103,82 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+    stop("Unknown type of a menu item.")
+  }
+}
+
+#' Create Semantic UI Menu
+#' 
+#' This creates a menu using Semantic UI.
+#' 
+#' @param type Type of the menu. Look at https://semantic-ui.com/collections/menu.html for all possiblities.
+#' @param ... Menu items to be created. Use menu_item function to create new menu item. Use right_menu to create a submenu on the right.
```
Comment (filipstachura) [inline R/dsl.R]: Please add one or two examples here.

**Comment by filipstachura** [comment]
That looks great! 💯  Just a few minor issues.

Please always try to set a meaningful title of the Pull Request. We can do better than "Mateusz uimenu" ;)

File: R/dsl.R [hunk 3]
```diff
@@ -88,105 +93,111 @@ uidropdown <- function(type = "", name = "", text = "", in_menu = "FALSE", ...,
...
+      tags$script(paste0("$('.ui.dropdown.", unique_dropdown_class, "').dropdown({", dropdown_functionality, "});"))
   )
 }
 
-
-
-#' Create Semantic UI Dropdown Choice Item
-#' 
-#' This creates a dropdown item using Semantic UI.
+#' Create Semantic UI Dropdown Menu
```
Comment (mateuszwyszynski) [inline R/dsl.R]: Dropdown Menu / Choices (?)

File: R/dsl.R [hunk 4]
```diff
@@ -10,59 +10,21 @@ uiicon <- function(type = "", ...) {
...
-    shiny::tags$script(paste0("$('.ui.dropdown.", unique_dropdown_class, "').dropdown().dropdown('set selected', '", value,"');"))
-  )
+#' Create Semantic UI label tag
+#' 
+#' This creates a label tag using Semantic UI.
+#' 
+#' @param type Type of the label. Look at https://semantic-ui.com/elements/label.html for all possibilities.
+#' @param ... Other arguments to be added such as content of the tag (text, icons) and/or attributes (style)
+uilabel <- function(type = "", ...) {
+  div(class = paste("ui label", type),
```
Comment (krystian8207) [inline R/dsl.R]: Many Semantic UI examples contain linkable labels (replacing `div` with `a`), maybe we could add this feature too in here?

File: R/dsl.R [hunk 5]
```diff
@@ -103,3 +65,228 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+#' @param type Type of the dropdown. Look at https://semantic-ui.com/modules/dropdown.html for all possibilities.
+#' @param name Unique name of the created dropdown.
+#' @param choices Use dropdown_menu to create set of choices, headers and dividers for the dropdown.
+#' @param ... Dropdown items to be created. Use dropdown_item to create new dropdown item.
+#' @param in_menu TRUE if the dropdown is a menu item. Default is FALSE.
+#' @param dropdown_specs A list of dropdown functionalities. Look at https://semantic-ui.com/modules/dropdown.html#link3 for all possibilities.
+#' @param value Default value of the dropdown. Useful when using selection dropdowns with default value selected.
+#' 
+#' @export
+uidropdown <- function(type = "", name, choices = dropdown_menu(), ..., in_menu = "FALSE", dropdown_specs = list(), value = NULL) {
```
Comment (krystian8207) [inline R/dsl.R]: Should be: in_menu = FALSE. "FALSE" is interpreted as bare string.

File: R/dsl.R [hunk 6]
```diff
@@ -103,3 +65,228 @@ generate_random_id <- function(prefix, id_length = 20) {
...
 }
+
+#' Create Semantic UI Dropdown
+#' 
+#' This creates a dropdown using Semantic UI.
+#' 
+#' @param type Type of the dropdown. Look at https://semantic-ui.com/modules/dropdown.html for all possibilities.
+#' @param name Unique name of the created dropdown.
+#' @param choices Use dropdown_menu to create set of choices, headers and dividers for the dropdown.
+#' @param ... Dropdown items to be created. Use dropdown_item to create new dropdown item.
```
Comment (krystian8207) [inline R/dsl.R]: Is there dropdown_item function?

File: R/dsl.R [hunk 7]
```diff
@@ -103,3 +65,228 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+  } else {
+    a(class = "item",
+      href = link,
+      `data-value` = name,
+      list(...)
+    )
+  }
+}
+
+#' Create Semantic UI Hedaer Item
```
Comment (krystian8207) [inline R/dsl.R]: Fix typo :)

File: R/dsl.R [hunk 8]
```diff
@@ -103,3 +65,228 @@ generate_random_id <- function(prefix, id_length = 20) {
...
+#' This creates a dropdown header item using Semantic UI.
+#' 
+#' @param ... Content of the header: text, icons, etc.
+#' 
+#' @export
+dropdown_header <- function(...) {
+  div(class = "header", ...)
+}
+
+#' Create Semantic UI Hedaer Item
```
Comment (krystian8207) [inline R/dsl.R]: Fix typo :)

**Comment by marekrogala** [comment]
@krystian8207 can you take over this PR and get it over the finish line?

File: R/dsl.R (hunk 6 above)
Comment (krystian8207) [reply R/dsl.R]: Fixed

File: R/dsl.R [hunk 9]
```diff
@@ -192,104 +307,169 @@ dropdown <- function(name,
...
+#' This creates a menu item using Semantic UI
+#'
+#' @param ... Content of the menu item: text, icons or labels to be displayed.
+#' @param item_feature If required, add additional item feature like 'active', 'header', etc.
+#' @param style Style of the item, e.g. "text-align: center".
+#' @param is_link If TRUE menu item is created with 'a' tag - ortherwise with 'div' tag (default).
+#' @param link If is_link is TRUE, determines to what the menu item should be linked. Ignored when is_link is FALSE.
+#'
+#' @export
+menu_item <- function(..., item_feature = "", style = NULL, is_link = FALSE, link = NULL) {
```
Comment (filipstachura) [inline R/dsl.R]: Please remove is_link from args. Please change link to href. When href != NULL => use tags$a.

File: R/dsl.R [hunk 10]
```diff
@@ -192,104 +307,169 @@ dropdown <- function(name,
...
 #' @export
-tabset <- function(tabs,
-                   id = generate_random_id("menu"),
-                   menu_class = "top attached tabular",
-                   tab_content_class = "bottom attached segment") {
-  identifiers <- replicate(length(tabs),
-                           list(id = generate_random_id("tab")),
-                           simplify = FALSE)
-  id_tabs <- purrr::map2(identifiers, tabs, ~ c(.x, .y))
+uimenu <- function(..., type = "", disable_ui_class = FALSE) {
```
Comment (filipstachura) [inline R/dsl.R]: Please remove `disable_ui_class`

**Pull request merged**